src/autopas/containers/hierarchicalGrids/HierarchicalGrids.h

```diff
@@ -0,0 +1,644 @@
+/**
+ * @file HierarchicalGrids.h
+ * @author hoppef
+ * @date 2023-07-10
+ * 
+ */
+
+#pragma once
+
+#include "autopas/cells/FullParticleCell.h"
+#include "autopas/containers/linkedCells/LinkedCells.h"
+#include "autopas/containers/CompatibleTraversals.h"
+#include "autopas/containers/LoadEstimators.h"
+#include "autopas/options/LoadEstimatorOption.h"
+
+#include "../applicationLibrary/discreteElementMethod/discreteElementMethodLibrary/DEMFunctor.h"
+
+#include <vector>
+#include <deque>
+
+namespace autopas {
+
+template <class Particle>
+class HierarchicalGrids : public ParticleContainerInterface<Particle> {
+ public:
+  /**
+   *  Type of the ParticleCell.
+   */
+  using ParticleCell = Particle; //FullParticleCell<Particle>;
+
+  /**
+   *  Type of the Particle.
+   */
+  using ParticleType = Particle; // debugging mit Piet
+  //using ParticleType = typename ParticleCell::ParticleType;
+
+  HierarchicalGrids (const std::array<double, 3> &boxMin, const std::array<double, 3> &boxMax, const double cutoff,
+                      const double skinPerTimestep, const unsigned int rebuildFrequency, const double cellSizeFactor = 1.0,
+                      LoadEstimatorOption loadEstimator = LoadEstimatorOption::squaredParticlesPerCell, 
+                      const unsigned int numberOfLevels = 1, const std::array<double, 2> particleRadiusRange = {0.0, 1.0})
+      : _boxMin(boxMin), _boxMax(boxMax), _cutoff(cutoff), _skinPerTimestep(skinPerTimestep), _rebuildFrequency(rebuildFrequency), 
+        _cellSizeFactor(cellSizeFactor), _loadEstimator(loadEstimator), 
+        _numberOfLevels(numberOfLevels), _particleRadiusRange(particleRadiusRange),
+        _carrierLevel(boxMin, boxMax, cutoff, skinPerTimestep, rebuildFrequency, cellSizeFactor, loadEstimator) {
+
+    // Loop over all hierarchy levels and create a Linked Cells container for each level.
+    //
+    // @todo !!! cellSizeFactor has to be adapted !!!
+    //
+
+
+
+      const double radiusMax = cutoff / 2.;
+      const double radiusSeg = radiusMax / numberOfLevels;
```

Comment:
are these assumptions reasonable? Wouldn't it make more sense to look at the number of particles in the system and allocate the levels according to their size and distribution? Aka make this more dynamic?

Reply:
Your suggestion would be the better solution and is a goal for the future. This was a first simplistic approach